rfcs/047-nested-columns.md

```diff
@@ -0,0 +1,189 @@
+* “I want to change the description for each level of nested columns”
+We expect these changes will enable our users to navigate and explore nested columns more easily and intuitively, encourage owners to add and maintain higher quality metadata, and help users to clearly understand the hierarchy of all types of nested columns.
+
+
+
+## Guide-level Explanation (aka Product Details)
+
+We have enhanced the way we represent nested columns, including the navigation and a layout change to showcase more metadata per nested columns.
+
+You can expand/collapse all nesting using the text button at top right corner.
+
+Please note that ⌄ icon is now used for expanding each level of nesting, rather than opening the column details within the list. Instead, when you click on the name of any column, you will see all its metadata in the dedicated right side panel. You can always close the panel to see more of the list.
+
+The hierarchy of nested objects without named fields (e.g. Array and Map) are now better specified and distinguished from nested objects with named fields (e.g. Struct).
+
+
+## UI/UX-level Explanation
+
+### Expand/collapse all + each level drill down
+
+![Default all expanded view](../assets/047/default-all-expanded-view.png)
+
+* As default, all nested columns will be expanded, showing the option for user to ‘Collapse all nested’
+* For most complex tables, this default behavior should still load fine, but in the edge case that there are prohibitively many columns to render (threshold will likely be ~1046), we may need to start with an all collapsed view for performance reasons.
+* Any nested columns (except for the top-level/parent column) will be a darker color to recede into the background, which can help users distinguish when quickly scanning the list.
+* The table header will be stick to the top at all times so users can always tell what the columns are as they scroll.
+
+
+
+![All collapsed view](../assets/047/all-collapsed-view.png)
+
+* When the user clicks the ‘Collapse all nested’ text button, the list will only show top-level or non-nested columns.
+* The up/down chevron that used to expand each column row to show full description and stats within the list area, will now change in functionality to expand/collapse each (one at a time, not all) level of the nested column. Therefore, only nested columns will have a chevron, and a column without any nesting under it will not have a chevron.
+
+
+
+![Each level drill down using chevron](../assets/047/each-level-drill-down-using-chevron.png)
+
+* Users can keep clicking on the chevron to expand each level of nesting if they want to  drill down the path without expanding all nesting at once.
+
+
+
+![Column details in right side panel](../assets/047/column-details-in-right-side-panel.png)
+
+* Clicking on a column name opens the right side panel to display all column details, including badges, description, and stats.
```

Comment:
should we mix changes from different rfcs? this is based on assumption that #46 is integrated in it's current form

Reply:
Yes, this is dependent on #46, we don't think we can do this without the other.

Reply:
Although this one is depending on #46, the reverse is not true. When we were drafting this, it started as one RFC but the content was clearly along two axes and we thought it would be more clear to separate.